File: 1078 - Occurrence After Bigram.cpp
class Solution {
public:
    vector<string> findOcurrences(string text, string first, string second) {
        vector<string> res;
        string a, b, word;
        text += ' ';
        for (char c : text) {
            if (c == ' ') {
                if (a == first && b == second) {
                    res.push_back(word);
                }
                a = b;
                b = word;
                word = "";
            } else {
                word += c;
            }
        }
        return res;
    }
};

File: 1078 - Occurrence After Bigram.js
/**
 * @param {string} text
 * @param {string} first
 * @param {string} second
 * @return {string[]}
 */
var findOcurrences = function(text, first, second) {
    let a = '', b = '', res = [];
    for (let word of text.split(' ')) {
        if (a == first && b == second) {
            res.push(word);
        }
        a = b;
        b = word;
    }
    return res;
};